### PR TITLE
[UR] Split the UNIFIED_RUNTIME_TAG to its own file

### PR DESCRIPTION
**Pull request by jsji**
With current repo setup, we need to update UNIFIED_RUNTIME_TAG
constantly. This is creating conflicts when doing merge from other
branches. To faciliate the automation to handle such conflicts,
we would prefer to split the tag into into own file, the automation
will then be triggereed whenever the file changes (implying UR tag
change).

### CODE REVIEW

**Comment by omirochn** [comment]
What about the second separate file for the same set for the xisa build?

**Comment by jsji** [comment]
> What about the second separate file for the same set for the xisa build?

Will be done in downstream branch only.

**Review by uditagarwal97 — COMMENTED** [review]
Can we also update the CODEOWNERS file to make unified-runtime-reviewers owners of the newly created file `sycl/cmake/modules/UnifiedRuntimeTAG.cmake`?

File: sycl/cmake/modules/UnifiedRuntimeTAG.cmake
Comment (kbenzie) [inline sycl/cmake/modules/UnifiedRuntimeTAG.cmake:1]: I think this should have the `Tag` suffix rather than `TAG`.

**Review by kbenzie — CHANGES_REQUESTED** [review]
I think this change makes sense. As mentioned by @uditagarwal97 the `.github/CODEOWNERS` file must be updated, for example as follows which includes my suggested filename change:

```diff
diff --git a/.github/CODEOWNERS b/.github/CODEOWNERS
index 085ad2fca16d..4e13514d2e9f 100644
--- a/.github/CODEOWNERS
+++ b/.github/CODEOWNERS
@@ -36,6 +36,7 @@ sycl/doc/extensions/ @intel/dpcpp-specification-reviewers

 # Unified Runtime
 sycl/cmake/modules/FetchUnifiedRuntime.cmake @intel/unified-runtime-reviewers
+sycl/cmake/modules/FetchUnifiedRuntimeTag.cmake @intel/unified-runtime-reviewers
 sycl/include/sycl/detail/ur.hpp @intel/unified-runtime-reviewers
 sycl/source/detail/posix_ur.cpp @intel/unified-runtime-reviewers
 sycl/source/detail/ur.cpp @intel/unified-runtime-reviewers
```

File: sycl/cmake/modules/UnifiedRuntimeTAG.cmake
Comment (jsji) [reply sycl/cmake/modules/UnifiedRuntimeTAG.cmake:1]: Sure.

**Comment by jsji** [comment]
> I think this change makes sense. As mentioned by @uditagarwal97 the `.github/CODEOWNERS` file must be updated, for example as follows which includes my suggested filename change:
> 
> ```diff
> diff --git a/.github/CODEOWNERS b/.github/CODEOWNERS
> --- a/.github/CODEOWNERS
> +++ b/.github/CODEOWNERS
> @@ -36,6 +36,7 @@ sycl/doc/extensions/ @intel/dpcpp-specification-reviewers
> 
>  # Unified Runtime
>  sycl/cmake/modules/FetchUnifiedRuntime.cmake @intel/unified-runtime-reviewers
> +sycl/cmake/modules/FetchUnifiedRuntimeTag.cmake @intel/unified-runtime-reviewers
>  sycl/include/sycl/detail/ur.hpp @intel/unified-runtime-reviewers
>  sycl/source/detail/posix_ur.cpp @intel/unified-runtime-reviewers
>  sycl/source/detail/ur.cpp @intel/unified-runtime-reviewers
> ```

Thanks @kbenzie and @uditagarwal97 .

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -36,6 +36,7 @@ sycl/doc/extensions/ @intel/dpcpp-specification-reviewers
 
 # Unified Runtime
 sycl/cmake/modules/FetchUnifiedRuntime.cmake @intel/unified-runtime-reviewers
+sycl/cmake/modules/FetchUnifiedRuntimeTag.cmake @intel/unified-runtime-reviewers
```
Comment (uditagarwal97) [inline .github/CODEOWNERS:39]:
```suggestion
sycl/cmake/modules/UnifiedRuntimeTag.cmake @intel/unified-runtime-reviewers
```

**Review by uditagarwal97 — APPROVED** [review]

**Review by kbenzie — APPROVED** [review]

**Comment by jsji** [comment]
@intel/llvm-gatekeepers Can someone help to merge this. Thanks.

**Comment by sarnex** [comment]
@jsji I must be missing it, but I don't see how this solves the problem. Won't merge conflict problem just be moved to `sycl/cmake/modules/UnifiedRuntimeTag.cmake` now?

**Comment by jsji** [comment]
> @jsji I must be missing it, but I don't see how this solves the problem. Won't merge conflict problem just be moved to `sycl/cmake/modules/UnifiedRuntimeTag.cmake` now?

Yes, you are right, this is not trying to remove the conflict in the tag itself, but to facilitate the automation to help with resolving the conflicts.

**Pull request merged**